diff --git a/servo_pkg/src/servo_mgr.cpp b/servo_pkg/src/servo_mgr.cpp
index e50be4c..64de37a 100644
--- a/servo_pkg/src/servo_mgr.cpp
+++ b/servo_pkg/src/servo_mgr.cpp
@@ -68,7 +68,7 @@ namespace PWM {
     void ServoMgr::servoSubscriber(const deepracer_interfaces_pkg::msg::ServoCtrlMsg::SharedPtr servoMsg) {
         auto setPWM = [&](auto &servo, auto value, auto type) {
             if (value < -1.0 || value > 1.0) {
-                RCLCPP_ERROR(logger_, "Invalid servo request: %d", value);
+                RCLCPP_ERROR(logger_, "Invalid servo request: %f", value);
                 return;
             }
 
diff --git a/servo_pkg/src/servo_node.cpp b/servo_pkg/src/servo_node.cpp
index 631af08..b83b786 100644
--- a/servo_pkg/src/servo_node.cpp
+++ b/servo_pkg/src/servo_node.cpp
@@ -62,7 +62,7 @@ int main(int argc, char **argv) {
                                                                                                  std::placeholders::_1),
                                                                                         rclcpp::SubscriptionOptions(),
                                                                                         rawPWMMsgStrategy);
-    auto servoCalServiceCbGrp_ = node->create_callback_group(rclcpp::callback_group::CallbackGroupType::Reentrant);                                                                                          
+    auto servoCalServiceCbGrp_ = node->create_callback_group(rclcpp::CallbackGroupType::Reentrant);
     rclcpp::Service<deepracer_interfaces_pkg::srv::SetCalibrationSrv>::SharedPtr servoCalService =
         node->create_service<deepracer_interfaces_pkg::srv::SetCalibrationSrv>(SET_CAL_SERVICE_NAME,
                                                                                std::bind(&PWM::ServoMgr::setCalHdl,
@@ -70,10 +70,10 @@ int main(int argc, char **argv) {
                                                                                std::placeholders::_1,
                                                                                std::placeholders::_2,
                                                                                std::placeholders::_3),
-                                                                               ::rmw_qos_profile_default,
+                                                                               rclcpp::QoS(rclcpp::SystemDefaultsQoS()),
                                                                                servoCalServiceCbGrp_);
 
-    auto setGPIOServiceCbGrp_ = node->create_callback_group(rclcpp::callback_group::CallbackGroupType::MutuallyExclusive);                                                                                          
+    auto setGPIOServiceCbGrp_ = node->create_callback_group(rclcpp::CallbackGroupType::MutuallyExclusive);
     rclcpp::Service<deepracer_interfaces_pkg::srv::ServoGPIOSrv>::SharedPtr setGPIOService =
         node->create_service<deepracer_interfaces_pkg::srv::ServoGPIOSrv>(SERVO_GPIO_SERVICE_NAME,
                                                                           std::bind(&PWM::ServoMgr::setGPIOHdl,
@@ -81,10 +81,10 @@ int main(int argc, char **argv) {
                                                                           std::placeholders::_1,
                                                                           std::placeholders::_2,
                                                                           std::placeholders::_3),
-                                                                          ::rmw_qos_profile_default,
+                                                                          rclcpp::QoS(rclcpp::SystemDefaultsQoS()),
                                                                           setGPIOServiceCbGrp_);
 
-    auto getCalServiceCbGrp_ = node->create_callback_group(rclcpp::callback_group::CallbackGroupType::Reentrant);                                                                                          
+    auto getCalServiceCbGrp_ = node->create_callback_group(rclcpp::CallbackGroupType::Reentrant);
     rclcpp::Service<deepracer_interfaces_pkg::srv::GetCalibrationSrv>::SharedPtr getCalService =
         node->create_service<deepracer_interfaces_pkg::srv::GetCalibrationSrv>(GET_CAL_SERVICE_NAME,
                                                                        std::bind(&PWM::ServoMgr::getCalHdl,
@@ -92,10 +92,10 @@ int main(int argc, char **argv) {
                                                                        std::placeholders::_1,
                                                                        std::placeholders::_2,
                                                                        std::placeholders::_3),
-                                                                       ::rmw_qos_profile_default,
+                                                                       rclcpp::QoS(rclcpp::SystemDefaultsQoS()),
                                                                        getCalServiceCbGrp_);
 
-    auto setLedCtrlCbGrp_ = node->create_callback_group(rclcpp::callback_group::CallbackGroupType::Reentrant);                                                                                          
+    auto setLedCtrlCbGrp_ = node->create_callback_group(rclcpp::CallbackGroupType::Reentrant);
     rclcpp::Service<deepracer_interfaces_pkg::srv::SetLedCtrlSrv>::SharedPtr setLedCtrlService =
         node->create_service<deepracer_interfaces_pkg::srv::SetLedCtrlSrv>(SET_LED_STATE_SERVICE_NAME,
                                                                            std::bind(&PWM::LedMgr::setLedCtrlHdl,
@@ -103,10 +103,10 @@ int main(int argc, char **argv) {
                                                                            std::placeholders::_1,
                                                                            std::placeholders::_2,
                                                                            std::placeholders::_3),
-                                                                           ::rmw_qos_profile_default,
+                                                                           rclcpp::QoS(rclcpp::SystemDefaultsQoS()),
                                                                            setLedCtrlCbGrp_);
 
-    auto getLedCtrlCbGrp_ = node->create_callback_group(rclcpp::callback_group::CallbackGroupType::Reentrant);                                                                                          
+    auto getLedCtrlCbGrp_ = node->create_callback_group(rclcpp::CallbackGroupType::Reentrant);
     rclcpp::Service<deepracer_interfaces_pkg::srv::GetLedCtrlSrv>::SharedPtr getLedCtrlService =
         node->create_service<deepracer_interfaces_pkg::srv::GetLedCtrlSrv>(GET_LED_STATE_SERVICE_NAME,
                                                                            std::bind(&PWM::LedMgr::getLedCtrlHdl,
@@ -114,7 +114,7 @@ int main(int argc, char **argv) {
                                                                            std::placeholders::_1,
                                                                            std::placeholders::_2,
                                                                            std::placeholders::_3),
-                                                                           ::rmw_qos_profile_default,
+                                                                           rclcpp::QoS(rclcpp::SystemDefaultsQoS()),
                                                                            getLedCtrlCbGrp_);
 
     RCLCPP_INFO(node->get_logger(), "servo_node started");
